### PR TITLE
Provide authentication in integration test cases

### PR DESCRIPTION
**Pull request by Ashish0410verma**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -80,11 +80,17 @@ Param(
     [Parameter(Mandatory=$true)]
     $Environment,
 
-    # The DbRequired for build database if it is true
-    [Parameter(Mandatory=$false)]
-    [string]
```
Comment (amitxagarwal) [inline build.ps1:85]: why this change here in this PR?

File: test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs [hunk 2]
```diff
@@ -17,12 +17,17 @@ public CitizenTests(IntegrationTestApplicationFactory factory)
             _factory = factory;
         }
 
-        [Fact(Skip = "Skipping the test cases for now")]
+        [SkipLocalFact]
```
Comment (amitxagarwal) [inline test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs:20]: why this new attribute?

File: test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Kmd.Momentum.Mea.Integration.Tests
+{
+    public class TokenGenerator : ITokenGenerator
```
Comment (amitxagarwal) [inline test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs:10]: what is the need for this class?

File: test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            Environment.GetEnvironmentVariable("KMD_MOMENTUM_MEA_ClientSecret");
+
+        public static string GetMeaScope() =>
+            Environment.GetEnvironmentVariable("KMD_MOMENTUM_MEA_Scope");
+
+        public readonly string tokenEndPoint = "clientCredentials/token?issuer=b2clogin.com&tenant=159";
+
+        public readonly string tokenEndPointAddress = "https://identity-api.kmdlogic.io/clientCredentials/token?issuer=b2clogin.com&tenant=159";
+
+        public async Task<string> GetToken()
```
Comment (amitxagarwal) [inline test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs:25]: this method is a part of MEA client. Can you refer the same method here instead of writing it again?

File: test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs (hunk 2 above)
Comment (Shrutika-Ingale) [reply test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs:20]: [SkipLocalFact] this makes better sense then the previous one.

File: test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs (hunk 3 above)
Comment (Shrutika-Ingale) [reply test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs:10]: To get the token for Integration Test cases while checking the APIs

File: test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs (hunk 4 above)
Comment (Shrutika-Ingale) [reply test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs:25]: Yea sure

Comment (Shrutika-Ingale) [reply test/Kmd.Momentum.Mea.Integration.Tests/TokenGenerator.cs:25]: its better if we have this part here itself, coz we need MEA client credentials for token and MeaClient has MCA Client Creds

File: build.ps1 [hunk 5]
```diff
@@ -242,10 +248,13 @@ try {
...
-            ($env:KMD_MOMENTUM_MEA_ClientSecret = $ClientSecret); 
-            ($env:KMD_MOMENTUM_MEA_ClientId = $ClientId); 
+            ($env:KMD_MOMENTUM_MEA_McaClientSecret = $ClientSecret); 
+            ($env:KMD_MOMENTUM_MEA_McaClientId = $ClientId); 
             ($env:KMD_MOMENTUM_MEA_McaApiUri = $McaApiUri); 
-            ($env:Scope = $Scope);
+            ($env:KMD_MOMENTUM_MEA_McaScope = $Scope);
+            ($env:KMD_MOMENTUM_MEA_ClientSecret = $MeaClientSecret); 
+            ($env:KMD_MOMENTUM_MEA_ClientId = $MeaClientId); 
+            ($env:KMD_MOMENTUM_MEA_Scope = $MeaScope);
```
[the remark this answers is not in the document]
Comment (Shrutika-Ingale) [reply build.ps1:262]: The intention was to get the client credentials from pipeline during build to run the test cases so that client credentials are kept secret and yet made available during build.

File: build.ps1 (hunk 1 above)
Comment (amitxagarwal) [reply build.ps1:85]: ?

File: test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs (hunk 2 above)
Comment (amitxagarwal) [reply test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs:20]: what is this for?

File: build.ps1 (hunk 5 above)
[the remark this answers is not in the document]
Comment (amitxagarwal) [reply build.ps1]: these credentials are for integration tests and just one set is sufficient which will provide access to any of the internal environments.

File: build.ps1 (hunk 1 above)
Comment (Shrutika-Ingale) [reply build.ps1:85]: Change made by Ashish

File: test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs (hunk 2 above)
Comment (Shrutika-Ingale) [reply test/Kmd.Momentum.Mea.Integration.Tests/Citizens/CitizenTests.cs:20]: To skip the particular test cases locally, we give this attribute to test cases.
